runtime/src/mm/cpp/ExceptionObjHolderTest.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#include <thread>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "Memory.h"
+#include "TestSupport.hpp"
+#include "ThreadData.hpp"
+#include "Types.h"
+
+using namespace kotlin;
+
+namespace {
+
+class ExceptionObjHolderTest : public ::testing::Test {
+public:
+    ~ExceptionObjHolderTest() {
+        auto& stableRefs = mm::StableRefRegistry::Instance();
+        stableRefs.ClearForTests();
```

Comment:
Because these tests are not very hermetic: even though separate threads are created, they still publish into a global, that's shared between tests. For now, this cleaning is required, but I'll be looking into decoupling, or (if it harms runtime performance / introduces many complications), alternatively, providing a test runner that'd automatically clean all the global state between test runs.